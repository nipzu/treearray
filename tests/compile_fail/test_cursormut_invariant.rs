use btreevec::CursorMut;

fn main() {}

pub fn bar<'c, 'a>(x: CursorMut<'c, &'static str>) -> CursorMut<'c, &'a str> {
    x
}
