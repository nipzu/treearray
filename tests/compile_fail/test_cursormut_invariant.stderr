error: lifetime may not live long enough
 --> tests/compile_fail/test_cursormut_invariant.rs:6:5
  |
5 | pub fn bar<'c, 'a>(x: CursorMut<'c, &'static str>) -> CursorMut<'c, &'a str> {
  |                -- lifetime `'a` defined here
6 |     x
  |     ^ returning this value requires that `'a` must outlive `'static`
  |
  = note: requirement occurs because of the type `bvec::CursorMut<'_, &str>`, which makes the generic argument `&str` invariant
  = note: the struct `bvec::CursorMut<'a, T>` is invariant over the parameter `T`
  = help: see <https://doc.rust-lang.org/nomicon/subtyping.html> for more information about variance
